content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -185,6 +184,33 @@ parallelism, for a variety of reasons:
 - The Job controller may throttle new Pod creation due to excessive previous pod failures in the same Job.
 - When a Pod is gracefully shut down, it takes time to stop.
 
+#### Completion Mode
+
+{{< feature-state for_k8s_version="v1.21" state="alpha" >}}
+
+{{< note >}}
+To be able to create Indexed Jobs, make sure to enable the `IndexedJob`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](docs/reference/command-line-tools-reference/kube-apiserver/)
+and the [controller manager](/docs/reference/command-line-tools-reference/kube-controller-manager/).
+{{< /note >}}
+
+Jobs with _fixed completion count_ - that is, jobs that have non null
+`.spec.completions` - can have a completion mode that is specified in `.spec.completionMode`:
+
+- `NonIndexed` (default): the Job is considered complete when there have been
+  `.spec.completions` successfully completed Pods. In other words, each Pod
+  completion is homologous to each other. Note that Jobs that have null
+  `.spec.completions` are implicitly `NonIndexed`.
+- `Indexed`: the Pods of a Job get an associated completion index from 0 to
+  `.spec.completions-1`, available in the annotation `batch.alpha.kubernetes.io/job-completion-index`.
```

Comment:
oh, I didn't notice that we used an alpha subdomain! The feature is already gated, and so I think using this a different annotation as it is being graduated is not necessary and will only cause unnecessary overhead for us maintaining the feature and job developers. If we agree on this, perhaps we can file an exception to remove that @liggitt?

Reply:
yes, please do not use alpha annotations. every time we've used that in the past it has attracted use we've had a very difficult time transitioning use to the non-alpha version

Reply:
I missed the policy, here is the fix kubernetes/kubernetes#100147